packages/components/src/templates/next/components/native/ListItem/ListItem.tsx

```diff
@@ -1,10 +1,23 @@
 import type { ListItemProps } from "~/interfaces"
-import Prose from "../Prose"
+import OrderedList from "../OrderedList"
+import Paragraph from "../Paragraph"
+import UnorderedList from "../UnorderedList"
 
 const ListItem = ({ content }: ListItemProps) => {
   return (
     <li className="my-5 pl-2 [&_p]:inline">
-      <Prose content={content} />
+      {content.map((item, index) => {
+        if (item.type === "paragraph") {
+          return <Paragraph key={index} {...item} />
+        } else if (item.type === "orderedList") {
+          return <OrderedList key={index} {...item} />
+        } else if (item.type === "unorderedList") {
+          return <UnorderedList key={index} {...item} />
+        } else {
+          const _: never = item
```

Comment:
type check - since we're declaring the type explicitly, if `typeof item !== never`, we will never be able to assign it to `_` and this will cause `npm run build` to fail with a type error

Reply:
Yup, learnt it last week that this is a good practice to ensure that you don't accidentally leave out a type.